projs/shadow/shadow-engine/core/inc/shadow/core/Time.h

```diff
@@ -1,18 +1,58 @@
 #pragma once
 
 #include "shadow/exports.h"
+#include <chrono>
 
-class Time {
+namespace SH {
 
-public:
-  static API int NOW;
-  static API int LAST;
+  // A RAII-operated timer.
+  // Instantiate to start the timer.
+  // Call timer.elapsed() to get the number of milliseconds since the timer was created.
+  // Call timer.swap() to reset the timer and get the time since instantiation.
+  class Timer {
 
-  static API double deltaTime;
-  static API double deltaTime_ms;
+  public:
+    std::chrono::high_resolution_clock::time_point timestamp = std::chrono::high_resolution_clock::now();
 
-  static API double timeSinceStart;
-  static API double startTime;
+    inline void record() {
+        timestamp = std::chrono::high_resolution_clock::now();
+    }
 
-  static void UpdateTime();
-};
+    inline double secondsSince(std::chrono::high_resolution_clock::time_point ts2) {
+        std::chrono::duration<double> span = std::chrono::duration_cast<std::chrono::duration<double>>(ts2 - timestamp);
+        return span.count();
+    }
+
+    inline double elapsedSeconds() {
+        return secondsSince(std::chrono::high_resolution_clock::now());
+    }
+
+    inline double elapsedMillis() {
+        return elapsedSeconds() * 1000.0f;
+    }
+
+    inline double elapsed() {
+        return elapsedMillis();
+    }
+
+    inline double swap() {
+        auto ts2 = std::chrono::high_resolution_clock::now();
+        auto elapsed = secondsSince(ts2);
+        timestamp = ts2;
+        return elapsed;
+    }
+
+    static size_t getTimestamp();

```

Comment:
Why is this `size_t` while the others are double?
Guessing this is the raw unix timestamp or similar, but a comment wouldn't hurt.

Reply:
It's a raw metric.

The others are in seconds/milliseconds, with high enough precision to have fractional parts (milliseconds/nanoseconds), but this one is directly from the OS, and both Linux and Windows return a size_t here. One is a Unix timestamp, one is a processor tick counter.

It's platform specific, so it's difficult to explain in the common section.

Reply:
If it is a platform depended thing and not even the same time scale than it needs a proper comment even more